tracing.go

```diff
@@ -257,7 +251,10 @@ func (s *Span) SetContext(key string, value Context) {
 
 // IsTransaction checks if the given span is a transaction.
 func (s *Span) IsTransaction() bool {
-	return s.isTransaction
+	if s.recorder != nil {
+		return s == s.recorder.root()
+	}
+	return false
```

Comment:
Can't we perform the same check as before?

```go
isTransaction: !hasParent
```

Reply:
Ah silly me. That's much simpler. I'll rework this all. :)